### PR TITLE
Added more unit tests for kube-scheduler.

### PR DESCRIPTION
**Pull request by tizhou86**
**What this PR does / why we need it**:
Added more unit tests for kube-scheduler.

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #
NONE

**Special notes for your reviewer**:
NONE

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: plugin/cmd/kube-scheduler/app/options/options_test.go [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+		"--kube-api-burst=80",
+		"--kube-api-content-type=application/vnd.kubernetes.protobuf",
+		"--kube-api-qps=40.0",
+		"--kubeconfig=/foo/bar",
+		"--leader-elect=true",
+		"--leader-elect-lease-duration=20s",
+		"--leader-elect-renew-deadline=15s",
+		"--leader-elect-resource-lock=endpoints",
+		"--leader-elect-retry-period=3s",
+		"--lock-object-name=foo",
```
Comment (bsalamat) [inline plugin/cmd/kube-scheduler/app/options/options_test.go]: Could you use something other than "foo" and "bar" repeatedly? The reason is that we want to make sure that one config is not taken for another one by mistake. For example, we want to ensure that "kubeconfig" file is not taken as "policy-config-file". The current values of the fields does not verify this.
You may want to use something like "test-lock-obj" here.

File: plugin/cmd/kube-scheduler/app/options/options_test.go [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+		"--lock-object-namespace=bar",
```
Comment (bsalamat) [inline plugin/cmd/kube-scheduler/app/options/options_test.go]: similarly, you may want to use "test-lock-ns" here.

File: plugin/cmd/kube-scheduler/app/options/options_test.go [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+		"--master=192.168.4.20",
+		"--policy-config-file=/foo/bar",
```
Comment (bsalamat) [inline plugin/cmd/kube-scheduler/app/options/options_test.go]: Please change "foo" and "bar" here and several other fields below.

File: plugin/cmd/kube-scheduler/app/options/options_test.go [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+*/
+
+package options
+
+import (
+	"reflect"
+	"testing"
+	"time"
+
+	"github.com/spf13/pflag"
```
Comment (bsalamat) [inline plugin/cmd/kube-scheduler/app/options/options_test.go]: Please put this at the end of the list.

**Review by bsalamat — COMMENTED** [review]
Thanks a lot, @tizhou86 for adding this test! I have several minor comments, but generally looks good.

File: plugin/cmd/kube-scheduler/app/options/options_test.go (hunk 1 above)
Comment (tizhou86) [reply plugin/cmd/kube-scheduler/app/options/options_test.go]: Sure, I'll do that.

**Comment by bsalamat** [comment]
/lgtm
/approve

Thanks, @tizhou86!

**Review by drinktee — APPROVED** [review]

**Pull request merged**